src/main/java/com/vroozi/repository/MaterialGroupDao.java

```diff
@@ -0,0 +1,42 @@
+package com.vroozi.repository;
+
+import com.mongodb.BasicDBObject;
+import com.vroozi.model.MaterialGroupMapping;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.aggregation.TypedAggregation;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class MaterialGroupDao {
+
+  @Autowired
+  private MongoTemplate mongoTemplate;
+
+  public Set<Integer> findAllUnitIds() {
+    TypedAggregation<MaterialGroupMapping> aggregation = Aggregation.newAggregation(
+        MaterialGroupMapping.class, Aggregation.group("unitId").push(
+            new BasicDBObject("_id", "$_id")).as("materialGroupMappings"));
```

Comment:
Distinct unitIds should be fetched from catalog collection because there's a possibility that a company hasn't uploaded material groups but catalogs are published